modules/shubin_m_numeric_systems/src/CustomNSNumber.cpp

```diff
@@ -0,0 +1,399 @@
+// Copyright 2024 Shubin Mikhail
+
+#include <stdexcept>
+#include "include/CustomNSNumber.hpp"
+
+CustomNSNumber::CustomNSNumber(number_t _number, size_t _num_sys) {
+  if (_num_sys < 2) {
+    throw std::invalid_argument("Incorrect numeric system");
+  }
+
+  num_sys = _num_sys;
+
+  number_t tmp = _number;
+  number_t number_t_num_sys = static_cast<number_t>(_num_sys);
+
+  if (tmp < 0) {
+    tmp *= -1;
+    negative = true;
+  }
+
+  while (tmp > 0) {
+    digits.push_back(tmp % number_t_num_sys);
+    tmp = tmp / number_t_num_sys;
+  }
+
+  if (digits.size() == 0) {
+    digits.push_back(0);
+  }
+}
+
+CustomNSNumber::CustomNSNumber(const CustomNSNumber& _num) {
+  num_sys = _num.num_sys;
+  negative = _num.negative;
+  digits = _num.digits;
+}
+
+CustomNSNumber::CustomNSNumber(CustomNSNumber&& _num) {
+  num_sys = _num.num_sys;
+  negative = _num.negative;
+  digits = _num.digits;
```

Comment:
Параметр `_num` является `l-value`, независимо от того, что его тип ссылка на `r-value`.
По этой причине в данном случае произойдет копирование, вместо ожидаемого перемещения.
```suggestion
  digits = std::move(_num.digits);
```
Поправьте также в операторе присваивания с перемещением

P.S: В вашем классе вовсе не обязательно иметь копирующие и перемещающие операции + деструктор, компилятор сгенерирует необходимые функции-члены самостоятельно